docker/importer/importer.py

```diff
@@ -452,6 +453,63 @@ def thread_init():
 
     logging.info("Finished processing bucket: %s", source_repo.name)
 
+  def _process_updates_rest(self, source_repo: osv.SourceRepository):
+    """Process updates from REST API."""
+    logging.info("Begin processing REST: %s", source_repo.name)
+    import_time_now = utcnow()
+    request = requests.head(source_repo.rest_api_url, timeout=60)
+    if request.status_code != 200:
+      logging.error('Failed to fetch REST API: %s', request.status_code)
+      return
+    last_modified = datetime.datetime.strptime(request.headers['Last-Modified'],
+                                               '%a, %d %b %Y %H:%M:%S %Z')
+    # Check whether endpoint has been modified since last update
+    if last_modified < source_repo.last_update_date:
+      logging.info('No changes since last update.')
+      return
+    request = requests.get(source_repo.rest_api_url, timeout=60)
+    # Get all vulnerabilities from the REST API. (CURL approach)
+    vulns = request.json()
+    vulns_to_update = []
+    # Create tasks for changed files.
+    for vuln in vulns:
+      import_failure_logs = []
+      last_modified = datetime.datetime.strptime(vuln['modified'],
+                                                 '%Y-%m-%dT%H:%M:%S.%fZ')
+
+      if last_modified < source_repo.last_update_date:
+        continue
+      try:
+        if source_repo.link[-1] != '/':
+          source_repo.link += '/'
+        single_request = requests.get(
+            source_repo.link + vuln['id'] + source_repo.extension, timeout=60)
+        single_vuln = single_request.json()
+        _ = osv.parse_vulnerability_from_dict(single_vuln, source_repo.key_path,
+                                              self._strict_validation)
+        vulns_to_update.append(single_vuln['id'])
+      except osv.sources.KeyPathError:
+        # Key path doesn't exist in the vulnerability.
+        # No need to log a full error, as this is expected result.
+        logging.info('Entry does not have an OSV entry: %s', vuln['id'])
+        continue
+      except Exception:
+        logging.error('Failed to parse %s', vuln['id'])
+        import_failure_logs.append('Failed to parse vulnerability "' +
+                                   vuln['id'] + '"')
+        continue
+    if vulns_to_update:
+      self._request_analysis_external(
+          source_repo, osv.sha256_bytes(source_repo.rest_api_url.encode()),
+          vulns_to_update)
```

Comment:
Hmm, I'm not sure passing this (a list) works here. WE need to request individual updates for every vuln to update, rather than this single one.

Reply:
This is updated now to request individual updates for every vuln. (Also fixes sha256 uniqueness problem)